runners/spark/src/main/java/org/apache/beam/runners/spark/stateful/SparkGroupAlsoByWindowViaWindowSet.java

```diff
@@ -479,23 +479,30 @@ private StateAndTimers(
 
   private static class OutputWindowedValueHolder<K, V>
       implements OutputWindowedValue<KV<K, Iterable<V>>> {
-    private List<WindowedValue<KV<K, Iterable<V>>>> windowedValues = new ArrayList<>();
+    private List<byte[]> windowedByteValues = new ArrayList<>();
+    private final Coder<WindowedValue<KV<K, Iterable<V>>>> wvKvIterCoder;
+
+    private OutputWindowedValueHolder(Coder<WindowedValue<KV<K, Iterable<V>>>> wvKvIterCoder) {
+      this.wvKvIterCoder = wvKvIterCoder;
+    }
 
     @Override
     public void outputWindowedValue(
         KV<K, Iterable<V>> output,
         Instant timestamp,
         Collection<? extends BoundedWindow> windows,
         PaneInfo pane) {
-      windowedValues.add(WindowedValue.of(output, timestamp, windows, pane));
+      windowedByteValues.add(
+          CoderHelpers.toByteArray(
```

Comment:
Would it make more sense here to override Java's serialization methods so we encode to bytes upon serialization instead of eagerly in the constructor?
This will also remove the need for a special member to hold the bytes.

Reply:
we can do it, probably it will be easier to do it like in SerializablePipelineOptions

Reply:
i will fix it as requested